### PR TITLE
hw28.1 is done

### PR DESCRIPTION
**Pull request by vladkuzya**

### CODE REVIEW

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect username or password"));
+
+        if (userFromDB.getPassword().equals(password)) {
+            return userFromDB;
+        }
+        throw new AuthenticationException("Incorrect username or password");
```
Comment (fyarkiy) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: I had the same - got a comment from mentor to merge so you have only in one place throw of exception

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("mate.academy");
+    private static final UserService userService = (UserService) injector
+            .getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (fyarkiy) [inline src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: remove empty line here

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        String url = req.getServletPath();
+        if (url.equals("/users/login") || url.equals("/users/registration")) {
+            chain.doFilter(req, resp);
+            return;
+        }
+
```
Comment (fyarkiy) [inline src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: empty line is not necessary here

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        if (userId == null || userService.getById(userId) == null) {
+            resp.sendRedirect(req.getContextPath() + "/users/login");
+            return;
+        }
+
```
Comment (fyarkiy) [inline src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: empty line

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        chain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (fyarkiy) [inline src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: empty line...

File: src/main/webapp/WEB-INF/views/users/login.jsp [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+<head>
+    <title>Login page</title>
+</head>
+<body>
+<h1 style="color:red">${errorMsg}</h1>
+<%@include file="/WEB-INF/views/header.jsp" %>
+<form method="post" action="${pageContext.request.contextPath}/users/login">
+    Your login: <input type="text" name="login">
+    Your password: <input type="password" name="psw">
+
```
Comment (fyarkiy) [inline src/main/webapp/WEB-INF/views/users/login.jsp]: empty line...

**Review by fyarkiy — APPROVED** [review]

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: done)

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java (hunk 2 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: done)

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java (hunk 3 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: done)

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java (hunk 4 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: done)

File: src/main/java/mate/academy/web/filter/AuthenticationFilter.java (hunk 5 above)
Comment (vladkuzya) [reply src/main/java/mate/academy/web/filter/AuthenticationFilter.java]: done)

File: src/main/webapp/WEB-INF/views/users/login.jsp (hunk 6 above)
Comment (vladkuzya) [reply src/main/webapp/WEB-INF/views/users/login.jsp]: done)

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("No such login exists"));
+
+        if (userFromDB.getPassword().equals(password)) {
+            return userFromDB;
+        }
+        throw new AuthenticationException("Incorrect password");
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: let's make both checks in one if() and throw exception only once with message: Incorrect login or password

Comment (vladkuzya) [reply src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: Done)

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**